code/modules/cm_marines/Xeno_Research/Xeno_Analyzation.dm

```diff
@@ -0,0 +1,257 @@
+
+/obj/machinery/computer/analyze_console/proc/SyncRDevices() //Makes sure it is properly sync'ed up with the devices attached to it (if any). Derived from rdconsole.dm
+	if(linked_chamber != null)
+		return
+	for(var/obj/machinery/alien_container/D in oview(3,src))
+		if(D.linked_console != null)
+			continue
+		linked_chamber = D
+		D.linked_console = src
+		return
+	return
+
+/obj/machinery/computer/analyze_console/Topic(href, href_list)				//Brutally teared from rdconsole.dm
+	if(..())
+		return
+
+	add_fingerprint(usr)
+
+	usr.set_interaction(src)
+	if(href_list["menu"]) //Switches menu screens. Converts a sent text string into a number. Saves a LOT of code.
+		var/temp_screen = text2num(href_list["menu"])
+		if(temp_screen <= 1.0 || src.allowed(usr) || emagged) //Unless you are making something, you need access.
+			screen = temp_screen
+		else
+			to_chat(usr, "Unauthorized Access.")
+
+	else if(href_list["find_device"]) //The R&D console looks for devices nearby to link up with.
+		screen = 0.0
+		spawn(20)
+			SyncRDevices()
+			screen = 1.3
+			updateUsrDialog()
+
+	else if(href_list["harvest"])
+		if(!linked_chamber)
+			updateUsrDialog()
+		else
+			if(!linked_chamber.occupant)
+				return
+			if(linked_chamber.occupant.autopsied == 1)
+				return
+			linked_chamber.occupant.autopsied = 1
+			screen = 0.3
+			spawn(300)
+				screen = 1.0
+				if(isXenoCrusher(linked_chamber.occupant))
+					new /obj/item/marineResearch/xenomorph/chitin/crusher(linked_chamber.loc)
+				else
+					new /obj/item/marineResearch/xenomorph/chitin(linked_chamber.loc)
+
+				new /obj/item/marineResearch/xenomorph/muscle(linked_chamber.loc)
+
+				switch(linked_chamber.occupant.caste_base_type)
+
+					if(/mob/living/carbon/Xenomorph/Warrior)
+						new /obj/item/marineResearch/xenomorph/muscle(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Ravager)
+						new /obj/item/marineResearch/xenomorph/muscle(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Drone)
+						new /obj/item/marineResearch/xenomorph/secretor(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Sentinel)
+						new /obj/item/marineResearch/xenomorph/acid_gland(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Spitter)
+						new /obj/item/marineResearch/xenomorph/acid_gland/spitter(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Hivelord)
+						new /obj/item/marineResearch/xenomorph/secretor(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/secretor/hivelord(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/secretor/hivelord(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Queen)
+						new /obj/item/marineResearch/xenomorph/acid_gland/spitter(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/secretor(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/secretor/hivelord(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/core(linked_chamber.loc)
```

Comment:
These could use a loot_list var or something on the xeno_caste datum.
And cut the length down to something like [this]( https://github.com/tgstation/tgstation/blob/55a768fef109af873291a4383e1b735a5e5969c5/code/modules/mob/living/simple_animal/simple_animal.dm#L303).